src/averageAges.js

```diff
@@ -37,7 +42,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = withChildren
+    ? people.filter((p) => p.sex === 'f'
+      && people.find((p1) => p1.mother === p.name))
```

Comment:
bad naming: it'd better to declare some variable like personName in outer function, in order access it in 'nested' function, so you can leave 'p' name (or even better 'person' in both functions). If you choose between shortness and readability - always choose readability
It would be better to use 'some' method instead of 'find', because it better expresses your intentions semantically: you want to check if at least one person meets the condition, not to find one